include/matx/transforms/fft.h

```diff
@@ -748,6 +749,53 @@ auto  GetFFTInputView([[maybe_unused]] OutputTensor &o,
   return i;
 }
 
+template <typename TensorOp>
+__MATX_INLINE__ auto getCufft1DSupportedTensor( const TensorOp &in, cudaStream_t stream) {
+
+  constexpr int RANK=TensorOp::Rank();
+
+  if constexpr ( !(is_tensor_view_v<TensorOp>)) {
+    return make_tensor<typename TensorOp::scalar_type>(in.Shape(), MATX_ASYNC_DEVICE_MEMORY, stream); 
+  } else {
+
+    bool supported = true;
+
+    // If there are any unsupported layouts for cufft add them here
+    
+    if (supported) {
+      return in;
+    } else {
+      return make_tensor<typename TensorOp::scalar_type>(in.Shape(), MATX_ASYNC_DEVICE_MEMORY, stream); 
+    }
+  }
+}
+
+template <typename TensorOp>
+__MATX_INLINE__ auto getCufft2DSupportedTensor( const TensorOp &in, cudaStream_t stream) {
+
+  constexpr int RANK=TensorOp::Rank();
+
+  if constexpr ( !is_tensor_view_v<TensorOp>) {
+    return make_tensor<typename TensorOp::scalar_type>(in.Shape(), MATX_ASYNC_DEVICE_MEMORY, stream); 
+  } else {
+    bool supported = true;
+
+    if ( in.Stride(RANK-2) != in.Stride(RANK-1) * in.Size(RANK-1)) {
```

Comment:
Maybe add a comment here so we can refer back to what is/isn't supported?

Reply:
fixed